lib/plug/adapters/cowboy2.ex

```diff
@@ -0,0 +1,306 @@
+  def https(plug, opts, cowboy_options \\ []) do
+    Application.ensure_all_started(:ssl)
+    run(:https, plug, opts, cowboy_options)
+  end
+
+  @doc """
+  Shutdowns the given reference.
+  """
+  def shutdown(ref) do
+    :cowboy.stop_listener(ref)
+  end
+
+  @doc """
+  Returns a child spec to be supervised by your application.
+
+  This function returns the old child specs used by early OTP
+  and Elixir versions. See `child_spec/1` for the Elixir v1.5
+  based child specifications.
+  """
+  def child_spec(scheme, plug, opts, cowboy_options \\ []) do
+    [ref, trans_opts, proto_opts] = args(scheme, plug, opts, cowboy_options)
+    cowboy_function = case scheme do
+      :http  -> :start_clear
+      :https -> :start_tls
+    end
+    cowboy_args = [ref, trans_opts, proto_opts]
+    {
+      {:ranch_listener_sup, ref},
+      {:cowboy, cowboy_function, cowboy_args},
+      :permanent, :infinity, :supervisor, [:ranch_listener_sup]
+    }
+  end
+
+  @doc """
+  A function for starting a Cowboy2 server under Elixir v1.5 supervisors.
+
+  It expects three options:
+
+    * `:scheme` - either `:http` or `:https`
+    * `:plug` - such as MyPlug or {MyPlug, plug_opts}
+    * `:options` - the server options as specified in the module documentation
+
+  ## Examples
+
+  Assuming your Plug module is named `MyApp` you can add it to your
+  supervision tree by using this function:
+
+      children = [
+        {Plug.Adapters.Cowboy2, scheme: :http, plug: MyApp, options: [port: 4040]}
+      ]
+
+      Supervisor.start_link(children, strategy: :one_for_one)
+
+  """
+  def child_spec(opts) do
+    scheme = Keyword.fetch!(opts, :scheme)
+    cowboy_opts = Keyword.fetch!(opts, :options)
+    {plug, plug_opts} =
+      case Keyword.fetch!(opts, :plug) do
+        {_, _} = tuple -> tuple
+        plug -> {plug, []}
+      end
+
+    {id, start, restart, shutdown, type, modules} =
+      child_spec(scheme, plug, plug_opts, cowboy_opts)
+
+    %{id: id, start: start, restart: restart,
+      shutdown: shutdown, type: type, modules: modules}
+  end
+
+  ## Helpers
+
+  @protocol_options [:timeout, :compress, :stream_handlers]
+
+  defp run(scheme, plug, opts, cowboy_options) do
+    case Application.ensure_all_started(:cowboy) do
+      {:ok, _} ->
+        :ok
+      {:error, {:cowboy, _}} ->
+        raise "could not start the Cowboy application. Please ensure it is listed as a dependency in your mix.exs"
+    end
+    start = case scheme do
+      :http  -> :start_clear
+      :https -> :start_tls
+      other  -> :erlang.error({:badarg, [other]})
+    end
+    apply(:cowboy, start, args(scheme, plug, opts, cowboy_options))
+  end
+
+  defp set_compress(cowboy_options) do
+    compress = Keyword.get(cowboy_options, :compress)
+    stream_handlers = Keyword.get(cowboy_options, :stream_handlers)
+    case {compress, stream_handlers} do
+      {true, nil} ->
+          Keyword.put_new(cowboy_options, :stream_handlers,
```

Comment:
Indentation is off.

Reply:
If you have the formatter set up, maybe we should run it on this file. But if you don't, no problem.